src/events/nats.js

```diff
@@ -55,6 +56,9 @@ const getPublishParams = (subject, msg, opt_reply, opt_callback, jsonConnectProp
         opt_callback_internal = opt_reply;
         opt_reply_internal = undefined;
     }
+    if (jsonConnectProperty && (process.env.EPSAGON_PROPAGATE_NATS_ID || '').toUpperCase() === 'TRUE') {
+        msg_internal.epsagon_id = uuid4();
```

Comment:
every message in nats already has an sid, no need to inject a new one (I can point you to the docs)